test/unit/helpers/multi.spec.ts

```diff
@@ -0,0 +1,62 @@
+import * as util from "../../util";
+import { InvalidMultiHelperFunctionUse } from "../../../src/transformation/utils/errors";
+
+test.each<[string, any]>([
+    ["let a; [a] = multi();", undefined],
+    ["const [a] = multi();", undefined],
+    ["const [a] = multi(1);", 1],
+    ["const ar = [1]; const [a] = multi(...ar);", 1],
+    ["const _ = null, [a] = multi(1);", 1],
+    ["let a; for (const [a] = multi(1, 2); false; 1) {}", undefined],
+    ["let a; for ([a] = multi(1, 2); false; 1) {}", 1],
+])("valid multi call and assign (%s)", (statement, result) => {
+    util.testModule`
+        ${statement}
+        export { a };
+    `
+        .setOptions({ types: ["typescript-to-lua/helpers"] })
+        .setReturnExport("a")
+        .expectToEqual(result);
+});
+
+test.each([
+    "multi",
+    "multi()",
+    "({ multi });",
+    "[] = multi()",
+    "const [] = multi();",
+    "const a = multi();",
```

Comment:
So, it doesn't support `tupleReturn`'s auto-packing? That's not necessarily a bad thing, but then it shouldn't be an array and `[...multi(...)]` should work

Reply:
I'd say `$multi` is meant more for clarifying function signatures in Lua, not as an alternate way to spread elements in an array

```ts
[...$multi(1, 2), 3] // [1, 2, 3]
// or write
[1, 2, 3]
```

Can you think of any use cases where `[...multi(...)]` should ideally work?

Reply:
> So, it doesn't support `tupleReturn`'s auto-packing?

In regards to avoiding auto-packing, this is so no value will be of type `MultiReturn`

Reply:
> Can you think of any use cases where `[...multi(...)]` should ideally work?

No, I'm perfectly fine with disallowing it. But in that case I also don't see much use in `const [a, b, c] = $multi(1, 2, 3)`. Should we disallow `$multi` everywhere except return statements?

Reply:
I think that's a good idea, pushed a change to allow `$multi` to only be used in ReturnStatements